### ISSUE TITLE
Test axum_factory::tests::test_supergraph_timeout fails if port 4000 is in use

### ISSUE THREAD

**Issue by glasser** [apollographql/router#3567]
The test axum_factory::tests::test_supergraph_timeout does not use an ephemeral port, so `cargo test` fails if port 4000 is in use (for example, if you just happen to be running a router with default configuration).

**Comment by abernix** [comment]
@nicholascioli Did you possibly fix this recently?

**Comment by nicholascioli** [comment]
Yes! [This PR](https://github.com/apollographql/router/pull/4722) makes the router bind to a random port for the integration tests. Is this issue happening even after the linked change?
